fendermint/vm/interpreter/src/fvm/gas/eip1559.rs

```diff
@@ -0,0 +1,108 @@
+use crate::fvm::gas::{Gas, GasMarket};
+use crate::fvm::state::{read_actor_state, FvmExecState};
+use crate::fvm::FvmMessage;
+use anyhow::anyhow;
+use fendermint_actor_gas_market::EIP1559GasState;
+use fendermint_vm_actor_interface::gas::GAS_ACTOR_ADDR;
+use fendermint_vm_actor_interface::system;
+use fvm_ipld_blockstore::Blockstore;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+type AtomicGas = AtomicU64;
+
+/// The gas market based on EIP1155
+#[derive(Default)]
+pub struct EIP1559GasMarket {
+    /// The block gas limit
+    block_gas_limit: AtomicGas,
+    /// The accumulated gas usage so far
+    block_gas_used: AtomicGas,
+}
+
+impl GasMarket for EIP1559GasMarket {
+    type State = EIP1559GasState;
+
+    fn reload_from_chain<DB: Blockstore + Clone + 'static>(
+        &self,
+        chain_state: &FvmExecState<DB>,
+    ) -> anyhow::Result<()> {
+        let state = get_state(chain_state)?;
```

Comment:
We need a query method in the `GasMarket` actor to fetch the current gas parameters (I called this `current()` in my proposal above), we should not dive into state directly as that is leaky.

Reply:
I added the method to the gas actor and query using the `getter` method. 

But in fendermint I think we dont really have to add the method to `GasMarket` because it limits the trait to have a `State` parameter, which `GasMarket` does not really need to be on chain. I think it's nicer to just keep this as an implementation detail.